pkg/logs/auditor/auditor.go

```diff
@@ -267,7 +272,29 @@ func (a *RegistryAuditor) flushRegistry() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(a.registryPath, mr, 0644)
+	f, err := os.CreateTemp(a.registryDirPath, a.registryTmpFile)
+	if err != nil {
+		return err
+	}
+	tmpName := f.Name()
+	defer func() {
+		if err != nil {
+			_ = f.Close()
+			_ = os.Remove(tmpName)
+		}
+	}()
+	if _, err := f.Write(mr); err != nil {
+		return err
+	}
+	if runtime.GOOS != "windows" {
+		if err := f.Chmod(0644); err != nil {
+			return err
+		}
+	}
+	if err := f.Close(); err != nil {
+		return err
+	}
+	return os.Rename(tmpName, a.registryPath)
```

Comment:
We should assign err here so the defer handler can cleanup if the rename fails:

```suggestion
	err = os.Rename(tmpName, a.registryPath)
	return err
```